lib/components/tiptap/image/ImageResizeComponent.tsx

```diff
@@ -1,39 +1,102 @@
 /* eslint-disable jsx-a11y/anchor-is-valid */
-import React from 'react'
-import { NodeViewWrapper } from '@tiptap/react'
-import { Resize } from './resizeIcon'
+import React, { useEffect, useState } from "react";
+import { NodeViewWrapper } from "@tiptap/react";
+import { Resize } from "./resizeIcon";
 
 export const ImageResizeComponent = (props: any) => {
-  const handler = (mouseDownEvent: React.MouseEvent<HTMLImageElement>) => {
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(false);
+
+  useEffect(() => {
+    const { id } = props.node.attrs;
+    const handleUpload = async () => {
+      if (props.editor) {
+        const { handleImageUpload } =
+          props.editor.options.extensions.find(
+            (ext: any) => ext.name === "imageResize"
+          )?.options || {};
+        if (handleImageUpload) {
+          try {
+            await handleImageUpload(id);
+            setLoading(false);
```

Comment:
Can you confirm that if `handleImageUpload` is not provided then there is no way for setLoading to get stuck at `true` (since setLoading(false) is inside if block